pkg/sdk/alert.go

```diff
@@ -0,0 +1,421 @@
+	schedule  *bool `ddl:"keyword" sql:"SCHEDULE"`
+	comment   *bool `ddl:"keyword" sql:"COMMENT"`
+}
+
+func (v *alerts) Alter(ctx context.Context, id SchemaObjectIdentifier, opts *AlterAlertOptions) error {
+	if opts == nil {
+		return errors.New("alter alert options cannot be empty")
+	}
+
+	opts.name = id
+	if err := opts.validate(); err != nil {
+		return err
+	}
+	sql, err := structToSQL(opts)
+	if err != nil {
+		return err
+	}
+	_, err = v.client.exec(ctx, sql)
+	return err
+}
+
+type DropAlertOptions struct {
+	drop  bool                   `ddl:"static" sql:"DROP"`  //lint:ignore U1000 This is used in the ddl tag
+	alert bool                   `ddl:"static" sql:"ALERT"` //lint:ignore U1000 This is used in the ddl tag
+	name  SchemaObjectIdentifier `ddl:"identifier"`
+}
+
+func (opts *DropAlertOptions) validate() error {
+	if !validObjectidentifier(opts.name) {
+		return ErrInvalidObjectIdentifier
+	}
+	return nil
+}
+
+func (v *alerts) Drop(ctx context.Context, id SchemaObjectIdentifier) error {
+	// alert drop does not support [IF EXISTS] so there are no drop options.
+	opts := &DropAlertOptions{
+		name: id,
+	}
+	if err := opts.validate(); err != nil {
+		return fmt.Errorf("validate alert options: %w", err)
+	}
+	sql, err := structToSQL(opts)
+	if err != nil {
+		return err
+	}
+	_, err = v.client.exec(ctx, sql)
+	if err != nil {
+		return err
+	}
+	return err
+}
+
+type ShowAlertOptions struct {
+	show   bool  `ddl:"static" sql:"SHOW"` //lint:ignore U1000 This is used in the ddl tag
+	Terse  *bool `ddl:"keyword" sql:"TERSE"`
+	alerts bool  `ddl:"static" sql:"ALERTS"` //lint:ignore U1000 This is used in the ddl tag
+
+	// optional
+	Like       *Like   `ddl:"keyword" sql:"LIKE"`
+	In         *In     `ddl:"keyword" sql:"IN"`
+	StartsWith *string `ddl:"parameter,no_equals,single_quotes" sql:"STARTS WITH"`
+	Limit      *int    `ddl:"parameter,no_equals" sql:"LIMIT"`
+}
+
+func (v *Alert) ID() SchemaObjectIdentifier {
+	return NewSchemaObjectIdentifier(v.DatabaseName, v.SchemaName, v.Name)
+}
+
+func (v *Alert) ObjectType() ObjectType {
+	return ObjectTypeAlert
+}
+
+type Alert struct {
+	CreatedOn    time.Time
+	Name         string
+	DatabaseName string
+	SchemaName   string
+	Owner        string
+	Comment      *string
+	Warehouse    string
+	Schedule     AlertSchedule
+	State        string
+	Condition    string
+	Action       string
+}
+
+type AlertSchedule interface {
+	alertSchedule()
+	String() string
+}
+
+func AlertScheduleFromString(s string) (AlertSchedule, bool) {
+	parts := strings.Fields(s)
+	if len(parts) == 2 && parts[len(parts)-1] == "MINUTE" {
+		return AlertScheduleInterval(toInt(parts[0])), true
+	} else if withoutCron := dropUsingCron(parts); len(withoutCron) == 6 {
+		return AlertScheduleCronExpression{
+			Expression: strings.Join(withoutCron[0:len(withoutCron)-1], " "),
+			TimeZone:   withoutCron[len(withoutCron)-1],
+		}, true
+	} else {
+		return nil, false
+	}
+}
+func dropUsingCron(parts []string) []string {
```

Comment:
lets try not have helper functions at the global level

Reply:
I'm not sure what \"global\" means in this context and where I should put it instead. 
I could put in some `helpers.go` inside sdk package, but this is an auxiliary function meant to be used only by `AlertScheduleFromString`

Reply:
I'm not sure what \"global\" means in this context and where I should put it instead. 
I could put in some `helpers.go` inside sdk package, but this is an auxiliary function meant to be used only by `AlertScheduleFromString`

Reply:
I'm not sure what \"global\" means in this context and where I should put it instead. 
I could put in some `helpers.go` inside sdk package, but this is an auxiliary function meant to be used only by `AlertScheduleFromString`

Reply:
I'm not sure what \"global\" means in this context and where I should put it instead. 
I could put in some `helpers.go` inside sdk package, but this is an auxiliary function meant to be used only by `AlertScheduleFromString`

Reply:
so because SDK is one big package, any function you create will be available for use anywhere in the package. Thats a lot of files. I would suggest either using an anonymous function or creating a "helper" struct and putting the functions in that struct so at least they are all in one place and not cluttering the namespace.